### ISSUE TITLE
Denormalized joins not included in `db->tree` result

### ISSUE THREAD

**Issue by anmonteiro** [omcljs/om#604]
```clojure
(def state
  {:items [{:id 1 :stuff [{:name "something"}]}]})

(defui Item
  static om/Ident
  (ident [this props]
    [:by-id (:id props)])
  static om/IQuery
  (query [this]
    [:id {:stuff [:name]}]))

(defui Items
  static om/IQuery
  (query [this]
    [{:items (om/get-query Item)}]))

(def norm (om/tree->db Items state true))
(om/db->tree (om/get-query Items) norm norm)
=> {:items [{:id 1, :stuff [{}]}]}
```

**Comment by jasonjckn** [comment]
+1

**Comment by awkay** [comment]
So, @anmonteiro and I discussed this at length. The discussion summary was basically:

Antonio: 
- Supporting this might encourage people to use denormalized state, which could lead to slow apps due to the lack of being able to do incremental queries.
- It might be hard for us to detect the right thing to do (ident detection is the only problem I'm aware of)

Me:
- Slow apps: they can always fix it by normalizing state.
- The query language "makes sense" for moving into objects that are tree-based, so if we *can* do what many user's expect (a join can traverse into a map by key instead of needing to follow an ident), then I see no reason not to. 

So, as we discussed this further the issue on detecting idents came up. I suggested that we could do a better job of detecting something being an ident by checking for the existence of the table indicated by the ident's keyword. If no such table even exists, then it isn't an ident...it is simply a vector. This potentially solves #637 

I'll discuss more about ident detection ideas there.

**Comment by TheSeldonPlan** [comment]
We ran into this problem also. We were using datomic enums, and using the pull syntax to get the db/ident on the enums. This works fine for single attr, but does not work with a many attr enum.
 
We used the workaround below:
```clj
(defcomponent Enum {}
  (ident [this props]
    [:enum/by-ident (:db/ident props)] )
  (query [_]
    [:db/ident]))

(defcomponent Schedule {}
  (ident [this props]
    [:schedule/by-id (:db/id props)])
  (query [this]
    [:db/id
     :schedule/title
     :schedule/emails
     :schedule/content
     :schedule/timezone
     {:schedule/frequency [:db/ident]}
     {:schedule/days (om/get-query Enum)}
     :schedule/hour
     :schedule/minute
     {:schedule/formats (om/get-query Enum)}
     :schedule/date]))
```
In this example frequency is a single attr but days and formats are many attrs. It seems kind of silly to have to normalize `[{:db/ident :schedule.formats/CSV}] => [[:enum/by-ident :schedule.formats/CSV]]`.

**Comment by anmonteiro** [comment]
this was also fixed in #725
